x-pack/plugins/security_solution/server/plugin.ts

```diff
@@ -116,6 +118,7 @@ export class Plugin implements ISecuritySolutionPlugin {
   private checkMetadataTransformsTask: CheckMetadataTransformsTask | undefined;
   private telemetryUsageCounter?: UsageCounter;
   private endpointContext: EndpointAppContext;
+  private riskEngineDataClient: RiskEngineDataClient | undefined;
```

Comment:
When will this be `undefined` ?

Reply:
Because we do call initialize this property on the setup method, but not the constructor, so it can be undefined before that.

We do this in setup, because we need to have `core` available